parity/db/rocksdb/mod.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright 2015-2018 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+extern crate kvdb_rocksdb;
+extern crate migration_rocksdb;
+
+use std::sync::Arc;
+use std::path::{Path, PathBuf};
+use ethcore::db::NUM_COLUMNS;
+use ethcore::client::{ClientConfig, DatabaseCompactionProfile};
+use kvdb::{KeyValueDB, KeyValueDBHandler};
+use self::kvdb_rocksdb::{Database, DatabaseConfig};
+
+use cache::CacheConfig;
+
+mod migration;
+mod helpers;
+
+pub use self::migration::migrate;
+
+// This function will only be used when compiling with secretstore feature.
+#[allow(dead_code)]
+pub fn open_secretstore_db(data_path: &str) -> Result<Arc<KeyValueDB>, String> {
+	let mut db_path = PathBuf::from(data_path);
+	db_path.push("db");
+	let db_path = db_path.to_str().ok_or_else(|| "Invalid secretstore path".to_string())?;
+	Ok(Arc::new(Database::open_default(&db_path).map_err(|e| format!("Error opening database: {:?}", e))?))
+}
+
+pub fn open_client_db(client_path: &Path, client_config: &ClientConfig) -> Result<Arc<KeyValueDB>, String> {
```

Comment:
Missing docs for public methods? Would be best to add `#![warn(missing_docs)]`

Reply:
I added docs for all public methods in `db/rocksdb/mod.rs`.

The weird thing is that the CLI already has `#![warn(missing_docs)]` in `main.rs` but I never saw any missing_docs warnings when building it.

Reply:
It has to be added to every crate, I suppose `db` crate might not have it.

Reply:
That's still a little bit weird though... `parity/db` is a mod not a separate crate.

Reply:
Okay I think I found out why. `#![warn(missing_docs)]` will only warn you if it's an exported item. Because none of `db` mod is exported, so there was no warning whether it has docs or not.